modules/ldsg.js

```
@@ -0,0 +1,147 @@
+    const modal = new u.Modal().addComponents(
+      u.ModalActionRow().addComponents([
+        new u.TextInput()
+          .setCustomId("text")
+          .setLabel("Reply")
+          .setStyle(Discord.TextInputStyle.Short)
+          .setRequired(true)
+          .setPlaceholder("Your reply for the user")
+      ])
+    ).setCustomId("suggestionReply").setTitle("Suggestion Reply");
+    let submitted;
+    let em;
+    let reply;
+    let member;
+    /** @type {String} */
+    let channel;
+    switch (int.customId) {
+      case "pa":
+        channel = u.sf.channels.publicaffairsForum;
+        break;
+      case "log":
+        channel = u.sf.channels.logisticsForum;
+        break;
+      case "ops":
+        channel = u.sf.channels.operationsForum;
+        break;
+      case "icarus":
+        channel = u.sf.channels.bottestingForums;
+        break;
+      case "mgmt":
+        channel = u.sf.channels.managementForum;
+        break;
+      case "reply":
+        await int.showModal(modal);
+        submitted = await int.awaitModalSubmit({ time: 5 * 60 * 1000, dispose: true }).catch(() => {
+          return null;
+        });
+        if (!submitted) {
+          return int.channel?.send("I fell asleep waiting for your input...");
+        }
+        await submitted.deferUpdate();
+        reply = submitted.fields.getTextInputValue("text");
+        em = u.embed({ author: int.user })
+          .setTitle("Suggestion feedback")
+          .addFields({ name: "Reply:", value: reply });
+        member = int.guild.members.cache.get(suggestion.embeds[0].fields.find(field => field.name === "User ID:")?.value ?? "");
+        if (!member) {
+          return int.channel?.send("I could not find that member");
+        }
+        try {
+          member.send({ embeds: [em] });
+          return int.channel?.send({ content: "Replied to user:", embeds: [em] });
+        } catch (e) {
+          u.errorHandler(e, int);
```

no need to report the error.